src/Transport/Topology/ITopology.cs

```diff
@@ -12,9 +16,87 @@ public interface ITopology {
         // TODO: CriticalError no longer passed in to MessagePumpFactory. Ensure that Core is doing pushMessages.OnCriticalError(error);
         Func<IPushMessages> GetMessagePumpFactory();
         Func<IDispatchMessages> GetDispatcherFactory();
+        StartupCheckResult ApplyPreStartupChecks();
         IManageSubscriptions GetSubscriptionManager();
         OutboundRoutingPolicy GetOutboundRoutingPolicy();
         bool HasNativePubSubSupport { get; }
         bool HasSupportForCentralizedPubSub { get;}
     }
+
+    // NOTE: where do we move this components?
+    public interface IStartupCheck
+    {
+        // NOTE: to make awaitable?
+        StartupCheckResult Apply();
```

Comment:
If startup check has to do IO (and it will), default should be async.
@yvesgoeleven any objections?